src/CiviEntityStorage.php

```diff
@@ -46,40 +33,35 @@ class CiviEntityStorage extends SqlContentEntityStorage implements DynamicallyFi
   protected $configFactory;
 
   /**
-   * Constructs a ContentEntityStorageBase object.
-   *
-   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
-   *   The entity type definition.
-   * @param \Drupal\Core\Database\Connection $database
-   *   The database connection.
-   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
-   *   The entity manager.
-   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
-   *   The cache backend to be used.
-   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
-   *   The language manager.
-   * @param \Drupal\civicrm_entity\CiviCrmApiInterface $civicrm_api
-   *   The CiviCRM API.
+   * TODO: Inject after forced 8.7
    */
-  public function __construct(EntityTypeInterface $entity_type, Connection $database, EntityManagerInterface $entity_manager, CacheBackendInterface $cache, LanguageManagerInterface $language_manager, CiviCrmApiInterface $civicrm_api, ConfigFactoryInterface $config_factory) {
-    parent::__construct($entity_type, $database, $entity_manager, $cache, $language_manager);
-    $this->civicrmApi = $civicrm_api;
-    $this->configFactory = $config_factory;
+  private function getCiviCrmApi() {
+    if (!$this->civicrmApi) {
+      $this->civicrmApi = \Drupal::service('civicrm_entity.api');
```

Comment:
Is there a way for us to use setter injection?

Reply:
I don't think so, like a `setCiviCrmAPI` method? I don't know when we would or could without worrying about more BC layers

Reply:
Yeah, ok... If this was a normal service, we could do it in the services.yml, but entity storage is instantiated with it's own special magic. I guess we don't even care about injection at this point because we don't have any tests anyway. :-) We can convert this back to injection when we need it.

Reply:
:) We do have some tests

Reply:
Do you know what we need to do to get Drupal.org's infrastructure to run them automatically when we commit?

Reply:
Ah, sorry! But I guess not unit tests if it doesn't matter if this is injected or not.

Reply:
I don't know how we could get DrupalCI to run, as it requires CiviCRM. We could on Travis. 

I was experimenting before for Webform CiviCRM /blob/travis-ci/.travis.yml

Reply:
@dsnopek correct, they're Kernel tests